app/HistoryExporter.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace App\Exporters;
+
+use App\Repositories\HistoryRepository;
+
+class HistoryExporter
+{
+    protected $data;
+    protected $historyRepository;
+
+    public function __construct()
+    {
+        $this->historyRepository = new HistoryRepository();
+    }
+
+    public function export($date = null)
+    {
+        $this->data = array_group($this->historyRepository->all(), 'date');
```

Comment:
let's implement new repository method `allGroupedBy`

```suggestion
        $data = $this->historyRepository->allGroupedBy('date');
```